### PR TITLE
Stat panel general cleanup + Improvements (Partial Bounty)

### PR DESCRIPTION
**Pull request by PowerfulBacon**
Was offered a £30 bounty by Floria in order to alter the listed turf stat panel to:
- Be sorted so that structures/machinery will be on top
- Concatenate similar items into 1 by putting (x30) instead.

I have done some additional improvements and changes to make it a bit nicer overall which were not part of this.

## About The Pull Request

Cleans up some of the elements of the stat panel which in the past have been a bit broken, now that I actually know react.

Firstly, prevents buttons from over-running the tab:
image

Makes the buttons extend the entire width of the screen:
image

Implements the ability to have multi-row buttons which support sub-buttons:
image
image

This also fixes an exploit where non-admins could delete SDQL queries that they know the ID of, as well as admins that don't have the correct permissions being able to delete SDQL queries.

Stat panel atoms will now extend the full width of the stat tab, have a tag showing what the thing is (which is coloured for better readability) and is sorted by tag.
image

The mouse cursor will now change when hovering over atoms in that menu to indicate that they can be dragged.

image

## Why It's Good For The Game

When I made the ticket panel, I thought that admins could have it open on their second screens however it is a mostly redundant panel due to the stat panel displaying the information better. Unfortunately, the claiming system is entirely dependant on that panel and not the stat panel, making it so its hard to see at a glance who has claimed a ticket. This fixes that.

Also cleans up some long outstanding issues with the stat panel/

## Testing Photographs and Procedure

See above.

## Changelog
:cl:
fix: Fixes stat panel buttons overrunning the screen on the MC tab.
fix: Enforces permission requirement on stat button presses.
add: Adds support for multi-row stat buttons which have sub-buttons.
/:cl:

### CODE REVIEW

**Review by itsmeow — APPROVED** [review]
I'm still of the opinion we should redesign the ticket manager and then directly embed this into the stat panel, but this works for now.

**Comment by PowerfulBacon** [comment]
It's possible, but easier said than done since the stat panel uses a unique system for getting its data meaning that you can't directly embed regular TGUI interfaces. There is also no ui act so you would need to have 2 different interfaces (How it works for the ticket messenging). This fits nicely into the main stat system as its entirely modular like the rest of the basic stat panels

File: tgui/packages/tgui-panel/stat/StatText.js [hunk 1]
```diff
@@ -61,19 +63,62 @@ export const StatTextText = (props, context) => {
...
+                'white-space': 'normal',
+              }}>
+              {text}
+            </Box>
+          </>
+        ) : (
+          <>
+            <Flex bold direction="row">
+              <Flex.Item grow={1}>{title}</Flex.Item>
+              {buttons.map((buttonInfo) => (
```
Comment (itsmeow) [inline tgui/packages/tgui-panel/stat/StatText.js:155]: extra buttons should be supported on non multirow elements

Comment (PowerfulBacon) [reply tgui/packages/tgui-panel/stat/StatText.js:155]: I was going to add this but its not easy to make this look good. Could probably just shove it in flex and let that deal with sizing.

**Comment by PowerfulBacon** [comment]
BUG: Humans always show as unknown in the mob panel, presumably due to some override icon things.

**Comment by PowerfulBacon** [comment]
"You" will now always display in the menu, regardless of if you are on the tile or not. This allows you to drag yourself onto other things and drag things onto yourself
image

**Comment by PowerfulBacon** [comment]
The limit of how many items it will search has also been raised.
image

**Comment by PowerfulBacon** [comment]
Now passes stress testing with no visible impact on the profiler despite large amounts of items on a turf.

image

image

image

**Review by itsmeow — APPROVED** [review]

**Pull request merged**